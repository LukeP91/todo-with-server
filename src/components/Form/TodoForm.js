import { Field, Form } from 'formik'
import { withStyles } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import PropTypes from 'prop-types'
import React from 'react'

import LabelSelect from './LabelSelect'
import TextInput from './TextInput'
import UserSelect from './UserSelect'

const styles = {
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    flexDirection: 'column',
  },
}

const TodoForm = ({ classes, labels, users }) => (
  <Form className={classes.container}>
    <Field component={TextInput} label="Title" name="title" />
    <Field component={TextInput} label="Description" name="description" />
    <Field component={UserSelect} label="User" name="userId" users={users} />
    <Field component={LabelSelect} label="Labels" labels={labels} name="labelsIds" />
    <Button color="primary" id="submit-button" type="submit" variant="contained">
      Submit
    </Button>
  </Form>
)

TodoForm.propTypes = {
  classes: PropTypes.object.isRequired,
  labels: PropTypes.array,
  users: PropTypes.array,
}

export default withStyles(styles)(TodoForm)
