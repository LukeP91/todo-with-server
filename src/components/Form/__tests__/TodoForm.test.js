import React from 'react'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'


import TodoForm from '../TodoForm'

describe('TodoForm', () => {
  xit('renders correctly', () => {
    const labels = [{ id: 1, value: 'Label1' }, { id: 2, value: 'Label2' }]
    const users = [{ id: 1, email: 'test@example.com', name: 'Test user' }]

    const renderer = new ShallowRenderer()
    renderer.render(<TodoForm labels={labels} users={users} />)
    const result = renderer.getRenderOutput()

    expect(result).toMatchSnapshot()
  })
})
