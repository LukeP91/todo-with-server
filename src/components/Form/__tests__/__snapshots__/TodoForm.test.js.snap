// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(TodoForm)
    labels={
      Array [
        Object {
          "id": 1,
          "value": "Label1",
        },
        Object {
          "id": 2,
          "value": "Label2",
        },
      ]
    }
    users={
      Array [
        Object {
          "email": "test@example.com",
          "id": 1,
          "name": "Test user",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container": "TodoForm-container-1",
      },
      "labels": Array [
        Object {
          "id": 1,
          "value": "Label1",
        },
        Object {
          "id": 2,
          "value": "Label2",
        },
      ],
      "users": Array [
        Object {
          "email": "test@example.com",
          "id": 1,
          "name": "Test user",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container": "TodoForm-container-1",
        },
        "labels": Array [
          Object {
            "id": 1,
            "value": "Label1",
          },
          Object {
            "id": 2,
            "value": "Label2",
          },
        ],
        "users": Array [
          Object {
            "email": "test@example.com",
            "id": 1,
            "name": "Test user",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
